### PR TITLE
Cherry-pick changes from `develop` -> `main`

### PR DESCRIPTION
**Pull request by baziorek**
Probably there were never full merge `develop` -> `main`, that is why it is that hard to do its now (I tried: https://github.com/hyperledger/iroha-python/pull/121). That is why instead of merge I'm just cherry-picking changes from `develop` to `main` branch.
_______
In `develop` branch was committed PR:
-------
1. https://github.com/hyperledger/iroha-python/pull/104 - serious problem fixed with python files generated by too new `protobuf`
2. https://github.com/hyperledger/iroha-python/pull/103 - small examples refactoring + added new example made by @iptelephony .
3. https://github.com/hyperledger/iroha-python/pull/107 - serious refactoring of examples + added two more examples
4. Added auto-restart to iroha: https://github.com/hyperledger/iroha-python/pull/122/commits/6781449d8081c0cdab80dc57a38f418734e109e1
______
In order to make its working in also Ubuntu 18.04 and Ubuntu 20.04 I needed to:
1. edit file: `setup.py` (requirements for versions only)
2. do the same as https://github.com/hyperledger/iroha-python/pull/104 but in Ubuntu 20.04 (but it also works in Ubuntu 22.04 now)

### CODE REVIEW

**Comment by baziorek** [comment]
Tests1: ubuntu 20.04 (docker):
----
```
$ git clone -b main https://github.com/baziorek/iroha-python.git
$ docker run -it  --name iroha_python_testing_release  --network=host -v $(pwd):/home/ --workdir /home  ubuntu:20.04

###### inside docker instance:
apt-get update && apt-get install python3-pip git -y

##### installing iroha-python with dependencies
pip3 install protobuf==3.15.0 grpcio_tools==1.21.1 six==1.12.0 # installed manually, because those versions are in: dev-requirements.txt file
pip3 install "git+https://github.com/baziorek/iroha-python.git@main"

##### running examples:
/home/iroha-python/examples/batch-example.py
# iroha 1.5 reset with -overwrite_ledger
/home/iroha-python/examples/tx-ordering.py
# iroha 1.5 reset with -overwrite_ledger
/home/iroha-python/examples/tx-example.py  
# iroha 1.5 reset with -overwrite_ledger
/home/iroha-python/examples/mst-example.py  
# iroha 1.5 reset with -overwrite_ledger
/home/iroha-python/examples/query-transactions.py  
```
____________
Test2 (Manjaro):
```
mkdir iroha_cherryPick && cd iroha_cherryPick
python3 -m venv venv/ && source venv/bin/activate

pip3 install protobuf==3.15.0 grpcio_tools==1.21.1 six==1.12.0 # installed manually, because those versions are in: dev-requirements.txt file
pip3 install "git+https://github.com/baziorek/iroha-python.git@main"

##### running examples:
/home/iroha-python/examples/tx-ordering.py
```

**Comment by baziorek** [comment]
I prepaired Dockerfile to allow easy-testing in various systems: https://gist.github.com/baziorek/490fe77786a32fc32f8295b5968a4d1a
_______________
Tested for (with the docker above):
-----
- [x] Ubuntu 18.04 - works perfect
- [x] Ubuntu 20.04 - works perfect
- [x] Ubuntu 22.04 - works perfect
- [x] Manjaro 20220807 - works perfect

File: examples/blocks-query.py [hunk 1]
```diff
@@ -4,12 +4,19 @@
...
+after calling function:  Iroha.blocks_query
+"""
+
 from iroha import Iroha, IrohaGrpc
 from iroha import IrohaCrypto
 import os
 import sys
 import time
 import uuid
+import grpc  # grpc.RpcError
```
Comment (appetrosyan) [inline examples/blocks-query.py:18]: I'll approve regardless, but if this is the only import, the `from grpc import RpcError` would have conveyed the same idea.

**Review by appetrosyan — APPROVED** [review]

File: examples/batch-example.py [hunk 2]
```diff
@@ -26,62 +41,81 @@
...
+    @wraps(func)
     def tracer(*args, **kwargs):
         name = func.__name__
-        print('\tEntering "{}"'.format(name))
+        stack_size = int(len(inspect.stack(0)) / 2)  # @wraps(func) is also increasing the size
+        indent = stack_size*'\t'
+        print(f'{indent}> Entering "{name}": args: {args}')
         result = func(*args, **kwargs)
-        print('\tLeaving "{}"'.format(name))
+        print(f'{indent}< Leaving "{name}"')
```
[the remark this answers is not in the document]
Comment (baziorek) [reply examples/batch-example.py]: I've corrected in 8 example files: 6736989e22f29577f70f77ec24fecf918f7051a5

File: examples/blocks-query.py (hunk 1 above)
Comment (baziorek) [reply examples/blocks-query.py]: Corrected: b36447fc3002b2d8b7617591a3363021126a443f

**Comment by baziorek** [comment]
@appetrosyan Thanks for Your review. I've corrected what You suggested in 2 commits:
- b36447fc3002b2d8b7617591a3363021126a443f
- 6736989e22f29577f70f77ec24fecf918f7051a5

**Review by iceseer — APPROVED** [review]

**Pull request merged**